server.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const User = require('./auth/userModel.js');
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
+
+
+
+mongoose.connect('mongodb://localhost/auth-i').then(() => {
+  console.log('*** Connected to database ***');
+});
+
+const server = express();
+
+
+server.use(express.json());
+
+const protected = (req, res, next) => {
+  if (req.session && req.session.username) next();
+  else res.status(401).json({ message: "Forbidden!"})
+}
+
+server.use(session({
+  secret: 'Q4gG3XU929munG39JPhl3HavYkhqTkdv9E3rjaDKtexrZRxfM7',
+  cookie: {
+    maxAge: 3600000
+  },
+  httpOnly: true,
+  secure: false,
+  resave: true,
+  saveUninitialized: false,
+  name: 'Mr. F',
+  store: new MongoStore({
+        url: 'mongodb://localhost/auth-i-sessiondata'
+  })
+}));
+
+server.get('/', (req, res,) => {
+  res.json({ api: 'running...' })
+});
+
+server.get('/view-counter', (req, res) => {
+	if (!req.session.viewCount) {
+		req.session.viewCount = 0;
+	}
+	req.session.viewCount++;
+	res.json({ viewCount: session.viewCount })
+})
+
+server.post('/api/login', (req,res) => {
+  const { username, password } = req.body
+  User.findOne({ username }, (err, user) => {
+    if (err) res.status(500).json({ err })
+    else user.comparePassword(password, (err, auth) => {
+      if (err) res.status(500).json({ err });
+      else if (auth) {
```

Comment:
Where did auth come from? I couldn't get this post request to work in Postman, and auth is the thing I can't figure out. Am I blind?

Reply:
auth comes from isMatch in my comparePassword method, when I do cb(null, isMatch).  isMatch, in turn, comes from the arguments given by bcrypt.compare to its callback function.

It's working fine for me... weird.  I have uncommitted changes, but nothing that should affect that route I don't think.  I'll revert and give it another whirl ...

Yeah, I get a "logged in successfully" message, and then my get requests on /api/users work.

Reply:
It's probably something on my end, thank you for confirming that it works.